unpublisher/src/unpublisher.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Client as ElasticClient,
+  errors as elasticErrors,
+} from "@elastic/elasticsearch";
+import log from "@weco/content-common/services/logging";
+import { Result } from "@elastic/elasticsearch/lib/api/types";
+
+type Clients = {
+  elastic: ElasticClient;
+};
+
+export const createUnpublisher =
+  (index: string) =>
+  async (clients: Clients, unpublishedDocuments: string[]) => {
+    // There is no way to tell from the payload whether the update is for unpublished
```

Comment:
An aside: at present the prismic webhook lambda that is pushing events to eventbridge passes them through unchanged which perpetuates this publish/unpublish ambiguity coming from the prismic side. The webhook lambda(s) could do more work to transform the event into something more custom we consume internally to be clearer about the purpose of the event and omit unnecessary fields.

Reply:
I don't necessarily disagree, but perhaps some useful context is that the webhook lambda was deliberately designed to pass the events through without a new model in order to separate the concerns of (a) a fixed endpoint for Prismic to call for a webhook and (b) pipeline/cluster-dependent handling of the events arriving via said webhook.

Reply:
I see. That makes sense, though i'd still favour having an internal event representation for the reasons above. You could have another step doing the transformation for internal services, [perhaps do that in EventBridge](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html) without another lambda (this is speculative, apart from a review of this PR).

Reply:
Not sure if I have understood correctly but also worth noting that it would also be impossible for the webhook Lambda to know whether the event is a publish/unpublish - that ambiguity can't be resolved without additional API calls

Reply:
We'd have a webhook per event type (configured from Prismic) i'd guess and then the lambda for each surfaced by the function URL is configured to know it's for that type of event (not being able to infer it from the event itself but we tell it what it is when we set it up).

Not blocking for merge, but interesting to think about what we might do in the future.

Reply:
@jamieparkinson points out we already do this here: https://github.com/wellcomecollection/content-api/blob/main/infrastructure/webhook_stack/events.tf#L9-L11 ✨